angularSTEP/src/app/recipe-page/recipe-page.component.ts

```diff
@@ -13,17 +18,128 @@ import {RecipeConverter} from '../recipe-converter';
 export class RecipePageComponent {
   id: string | null = this.route.snapshot.paramMap.get('id');
   pageRecipe!: Recipe;
+  baseRecipe!: Recipe;
+  currentUser!: User;
+  baseUser!: User;
+  loggedIn: boolean = false;
 
-  constructor(private db: AngularFirestore,
-              private route: ActivatedRoute,) {
-                this.setRecipeData();
+  constructor(
+    private db: AngularFirestore,
+    private route: ActivatedRoute,
+    private fAuth: AngularFireAuth,
+    private dialog: MatDialog,
+    private router: Router,
+  ) {
+    this.setPageData();
+    this.fAuth.onAuthStateChanged(auth => {
+      if (auth) {
+        this.setPageData();
+        this.setUserData(auth.uid);
+        this.loggedIn = true;
+      } else {
+        this.setPageData();
+        this.loggedIn = false;
+      }
+    });
   }
 
-  async setRecipeData() {
-    const postRecipe = await this.db.doc('/recipes/' + this.id + '/').ref.withConverter(new RecipeConverter().recipeConverter).get();
+  async setPageData() {
+    const postRecipe = await this.db.doc('/recipes/' + this.id + '/')
+      .ref.withConverter(new RecipeConverter().recipeConverter).get();
     const recipeClassData = postRecipe.data();
+
     if (recipeClassData) {
+
+
+      if (recipeClassData.baseUploaderUid) {
+        const postBaseUser = await this.db.doc('/users/' + recipeClassData.baseUploaderUid + '/')
+          .ref.withConverter(new Converter().userConverter).get();
+        const baseUserData = postBaseUser.data();
+      
+        if (baseUserData) {
+          this.baseUser = baseUserData;
+
+          const postBaseRecipe = await this.db.doc('/recipes/' + recipeClassData.baseRecipeId + '/')
+            .ref.withConverter(new RecipeConverter().recipeConverter).get();
+          const baseRecipeData = postBaseRecipe.data();
+
+          if (baseRecipeData) {
+            this.baseRecipe = baseRecipeData;
+          }
+        }
+      }
+
       this.pageRecipe = recipeClassData;
     }
-  } 
+  }
+
+  async setUserData(uid: string) {
+    const postUser = await this.db.doc('/users/' + uid + '/')
+      .ref.withConverter(new Converter().userConverter).get();
+    if (postUser && postUser.data()) {
+      this.currentUser = postUser.data()!;
+    }
+  }
+
+  objToMap(obj: any): Map<string, number> {
+    const mp = new Map();
+    Object.keys(obj).forEach(k => {mp.set(k, obj[k])});
+    return mp;
+  }
+
+  addItem(item: string) {
+    var currentShoppingList: Map<string, number> = this.objToMap(this.currentUser.shoppingList);
+
+    const dialogConfig = new MatDialogConfig();
+
+    dialogConfig.disableClose = true;
+    dialogConfig.autoFocus = true;
+    dialogConfig.height = 'auto';
+    dialogConfig.width = 'auto';
+    dialogConfig.data = {
+      selectedItem: item,
+      add: true,
```

Comment:
Okay I also believe this stuff is from the other PR. I would merge the other one and then make sure this one has the same changes.